### PR TITLE
[Error] Add error message when the number of elements in kernel arguments exceed

### PR DESCRIPTION
**Pull request by mzmzm**
Add error message when the number of elements in kernel arguments exceeds.

### CODE REVIEW

File: python/taichi/lang/kernel_impl.py [hunk 1]
```diff
@@ -608,6 +608,20 @@ def func__(*args):
             if not self.is_grad and self.runtime.target_tape and not self.runtime.grad_replaced:
                 self.runtime.target_tape.insert(self, args)
 
+            if actual_argument_slot > 8 and (
+                    _ti_core.arch_name(impl.current_cfg().arch) == 'opengl'
+                    or _ti_core.arch_name(impl.current_cfg().arch) == 'cc'):
+                raise RuntimeError(
```
Comment (strongoier) [inline python/taichi/lang/kernel_impl.py:614]: `TaichiRuntimeError`

**Review by strongoier — CHANGES_REQUESTED** [review]
Add some tests :-)

File: python/taichi/lang/kernel_impl.py (hunk 1 above)
Comment (mzmzm) [reply python/taichi/lang/kernel_impl.py]: fix

File: tests/python/test_argument.py [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+        return a + b + c + d + e + f + g + h
+
+    assert foo1(1, 2, 3, 4, 5, 6, 7, 8) == 36
+
+    @ti.kernel
+    def foo2(a: ti.i32, b: ti.i32, c: ti.i32, d: ti.i32, e: ti.i32, f: ti.i32,
+             g: ti.i32, h: ti.i32, i: ti.i32) -> ti.i32:
+        return a + b + c + d + e + f + g + h + i
+
+    with pytest.raises(ti.TaichiRuntimeError):
```
Comment (strongoier) [inline tests/python/test_argument.py:21]: Use `match=xxx` to make the validation more explicit.

Comment (mzmzm) [reply tests/python/test_argument.py]: fix

File: python/taichi/lang/kernel_impl.py [hunk 3]
```diff
@@ -608,6 +608,20 @@ def func__(*args):
             if not self.is_grad and self.runtime.target_tape and not self.runtime.grad_replaced:
                 self.runtime.target_tape.insert(self, args)
 
+            if actual_argument_slot > 8 and (
+                    _ti_core.arch_name(impl.current_cfg().arch) == 'opengl'
```
Comment (strongoier) [inline python/taichi/lang/kernel_impl.py:612]: Compare arch with `ti.opengl` directly?

File: tests/python/test_argument.py [hunk 4]
```diff
@@ -0,0 +1,183 @@
...
+            ti.TaichiRuntimeError,
+            match=
+            "The number of elements in kernel arguments is too big! Do not exceed 8 on cc backend."
+    ):
+        foo2(1, 2, 3, 4, 5, 6, 7, 8, 9)
+
+
+@test_utils.test(arch=ti.cuda)
+def test_exceed_max_64():
+    N = 64
```
Comment (strongoier) [inline tests/python/test_argument.py:53]: Try not to separate these tests..

File: python/taichi/lang/kernel_impl.py (hunk 3 above)
Comment (mzmzm) [reply python/taichi/lang/kernel_impl.py]: fix

File: tests/python/test_argument.py (hunk 4 above)
Comment (mzmzm) [reply tests/python/test_argument.py]: fix

**Review by strongoier — APPROVED** [review]
LGTM!

**Pull request merged**